news/6419.bugfix

```diff
@@ -0,0 +1 @@
+Fix NameError when handling InvalidRequirement in install_req_from_req_string.
```

Comment:
Do you happen to know if this error can be hit from a pip invocation from the CLI? It would be best if the news entry can be phrased without referring to constructs in the code (i.e. so only what end-users would see). Would an end-user see a `NameError`? (Also, code things like exception classes should be surrounded by double backticks both before and after as this is reStructuredText.)

Reply:
I wasn't able to trigger it and there aren't any open issues about it, so I'm not sure. Will make the news fragment rST

Reply:
Okay, then I would just say something like "Fix NameError when handling an invalid requirement" (without referencing `install_req_from_req_string` or the `InvalidRequirement` class).